sale_shipping_info_helper/models/sale_order.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2017-To Day Akretion
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# @author: Mourad EL HADJ MIMOUNE <mourad.elhadj.mimoune@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    shipping_amount_total = fields.Float(
+        compute="_compute_shipping", digits="Account", store=True
+    )
+    shipping_amount_untaxed = fields.Float(
+        compute="_compute_shipping", digits="Account", store=True
+    )
+    shipping_amount_tax = fields.Float(
+        compute="_compute_shipping", digits="Account", store=True
+    )
+    item_amount_total = fields.Float(
+        compute="_compute_shipping", digits="Account", store=True
+    )
+    item_amount_untaxed = fields.Float(
+        compute="_compute_shipping", digits="Account", store=True
+    )
+    item_amount_tax = fields.Float(
+        compute="_compute_shipping", digits="Account", store=True
+    )
+
+    @api.depends("amount_total", "amount_untaxed")
+    def _compute_shipping(self):
+        for record in self:
+            record.shipping_amount_untaxed = (
+                record.shipping_amount_total
+            ) = record.shipping_amount_tax = 0
+            for line in record.order_line:
+                if not line.is_delivery:
+                    continue
+                record.update(
```

Comment:
If we have partial DO and shipped later, then we might have multiple shipping line, so we need to append the values of each shipping line right?

Reply:
Indeed you can have several shipping lines.

Thanks